nipype/algorithms/tests/test_modelgen.py

```diff
@@ -17,7 +17,7 @@
 
 
 def test_modelgen1(tmpdir):
-    tempdir = str(tmpdir)
+    tempdir = tmpdir.strpath
     filename1 = os.path.join(tempdir, 'test1.nii')
```

Comment:
so this could be simply `tmpdir.join('test.nii')`

Reply:
ok, can try to find those places automatically

Reply:
but `py.path.local` won't work everywhere, I'm recalling that I did have some problems with it when I was converting tests to pytest, and there were places where string worked better.   I can see that that at least for `nibabel.save` I'll have to keep strings.

Reply:
i meant places where the methods of `py.path.local` make sense, such as join. it's not a direct replacement for a string in all cases.

Reply:
my only point was that this test will not work if I change join part to `tmpdir.join('test.nii')`, it has to be string. 
Can always do `tmpdir.join('test.nii').strpath` if this is preferred.